### ISSUE TITLE
Event Dispatcher

### ISSUE THREAD

**Issue by namaless** [CouscousPHP/Couscous#180]
Hi all, the event dispatcher already in mind to add in the project?

I need to change the markdown parser (add block types and more) and with a good place for events I think is easy and not need to replace the class or anything in the original code, what you think?

**Comment by bobmulder** [comment]
I came up with this idea before, like #150...

**Comment by namaless** [comment]
Yeah I have already read your work (great) but I wanna know the situation from the main developers, but I don't think they wanna do it.

I still working with a complete rework of this, for a moment a clean job, after check if is what I want I can try to share my experience here but I don't think they want change the plans...

We need to waiting :)
